packages/react-native/ReactCommon/react/performance/timeline/PerformanceObserver.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "PerformanceObserver.h"
+#include "PerformanceObserverRegistry.h"
+#include "PerformanceEntryReporter.h"
+
+namespace facebook::react {
+
+PerformanceObserver::~PerformanceObserver() {
+  if (auto registry = registry_.lock()) {
+    registry->removeObserver(*this);
```

Comment:
The asymmetry between adding the observer and removing it feels wrong. I think we should also add the observer in the constructor (or in the `observe` method the first time it's called).